File: leetcode-go/simplify-path/simplify-path_test.go
package simplify_path

import "testing"

func TestSimplifyPathExample1(t *testing.T) {
	path := "/home/"
	ans := simplifyPath(path)
	rightAns := "/home"
	if ans != rightAns {
		t.Error("right ans = ", rightAns, ", current ans = ", ans)
	}
}

func TestSimplifyPathExample2(t *testing.T) {
	path := "/../"
	ans := simplifyPath(path)
	rightAns := "/"
	if ans != rightAns {
		t.Error("right ans = ", rightAns, ", current ans = ", ans)
	}
}

func TestSimplifyPathExample3(t *testing.T) {
	path := "/home//foo/"
	ans := simplifyPath(path)
	rightAns := "/home/foo"
	if ans != rightAns {
		t.Error("right ans = ", rightAns, ", current ans = ", ans)
	}
}

func TestSimplifyPathExample4(t *testing.T) {
	path := "/a/./b/../../c/"
	ans := simplifyPath(path)
	rightAns := "/c"
	if ans != rightAns {
		t.Error("right ans = ", rightAns, ", current ans = ", ans)
	}
}

File: leetcode-cpp/simplify-path/simplify-path-test.cpp
//
// Created by smile on 2022/1/9.
//

#include <gtest/gtest.h>
#include "simplify-path.cpp"

namespace {
    TEST(simplifyPathTest, Example1) {      // NOLINT
        auto path = "/../";
        auto ans = Solution::simplifyPath(path);
        decltype(ans) correctAns = "/";
        ASSERT_EQ(ans, correctAns);
    }

    TEST(simplifyPathTest, Example2) {      // NOLINT
        auto path = "/home/";
        auto ans = Solution::simplifyPath(path);
        decltype(ans) correctAns = "/home";
        ASSERT_EQ(ans, correctAns);
    }

    TEST(simplifyPathTest, Example3) {      // NOLINT
        auto path = "/home//foo/";
        auto ans = Solution::simplifyPath(path);
        decltype(ans) correctAns = "/home/foo";
        ASSERT_EQ(ans, correctAns);
    }

    TEST(simplifyPathTest, Test0) {      // NOLINT
        auto path = "/a/./b/../../c/";
        auto ans = Solution::simplifyPath(path);
        decltype(ans) correctAns = "/c";
        ASSERT_EQ(ans, correctAns);
    }

    TEST(simplifyPathTest, Test1) {      // NOLINT
        auto path = "/a//b////c/d//././/..";
        auto ans = Solution::simplifyPath(path);
        decltype(ans) correctAns = "/a/b/c";
        ASSERT_EQ(ans, correctAns);
    }
}

int main(int argc, char *argv[]) {
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}
